lte/gateway/c/core/oai/tasks/mme_app/mme_config.c

```diff
@@ -308,6 +308,116 @@ void mme_config_exit(void) {
   }
 }
 
+/****************************************************************************
+ **                                                                        **
+ ** Name:        copy_plmn_from_config()                                   **
+ **                                                                        **
+ ** Description: Copies the tai list configuration to partial tai list.    **
+ **                                                                        **
+ ** Inputs:  served_tai:        Served tai constructed from MME config.    **
+ **          index:             Index to used on served_tai                **
+ **                  Others:    None                                       **
+ **                                                                        **
+ ** Outputs:     None                                                      **
+ **          plmn:   PLMN filled from served_tai                           **
+ **                  Return:    RETURNok, RETURNerror                      **
+ **                  Others:                                               **
+ **                                                                        **
+ ***************************************************************************/
+int copy_plmn_from_config(
+    const served_tai_t* served_tai, int index, plmn_t* plmn) {
+  plmn->mcc_digit1 = (served_tai->plmn_mcc[index] / 100) % 10;
+  plmn->mcc_digit2 = (served_tai->plmn_mcc[index] / 10) % 10;
+  plmn->mcc_digit3 = served_tai->plmn_mcc[index] % 10;
+  if (served_tai->plmn_mnc_len[index] == 2) {
+    plmn->mnc_digit1 = (served_tai->plmn_mnc[index] / 10) % 10;
+    plmn->mnc_digit2 = served_tai->plmn_mnc[index] % 10;
+    plmn->mnc_digit3 = 0xf;
+  } else if (served_tai->plmn_mnc_len[index] == 3) {
+    plmn->mnc_digit1 = (served_tai->plmn_mnc[index] / 100) % 10;
+    plmn->mnc_digit2 = (served_tai->plmn_mnc[index] / 10) % 10;
+    plmn->mnc_digit3 = served_tai->plmn_mnc[index] % 10;
+  } else {
+    return RETURNerror;
+  }
+  return RETURNok;
+}
+
+void create_partial_lists(mme_config_t* config_pP) {
+  uint8_t elem_idx = 0, list_idx = 0;
+  /* Copy TAIs from served_tai to partial lists. If there are more that 16 TAIs,
+   * add the TAIs to a new partial list
+   */
+  uint8_t size = config_pP->served_tai.nb_tai > MAX_TAI_SUPPORTED ?
+                     MAX_TAI_SUPPORTED :
+                     config_pP->served_tai.nb_tai;
+  config_pP->partial_list = calloc(size, size * sizeof(partial_list_t));
```

Comment:
Isn't it calloc(size, sizeof(partial_list_t)) or calloc(1, size * sizeof(partial_list_t)) ?

Reply:
+1